### PR TITLE
[now-routing-utils] Add validation for `handle:hit` and `handle:miss`

### PR DESCRIPTION
**Pull request by styfle**
This PR adds initial support for `handle: hit` and `handle: miss` routes.

### CODE REVIEW

File: packages/now-routing-utils/test/index.spec.js [hunk 1]
```diff
@@ -54,6 +54,10 @@ describe('normalizeRoutes', () => {
       },
       { handle: 'filesystem' },
       { src: '^/(?<slug>[^/]+)$', dest: 'blog?slug=$slug' },
+      { handle: 'hit' },
+      { src: '^/hit-me$', dest: '/api/hit-miss' },
```
Comment (dav-is) [inline packages/now-routing-utils/test/index.spec.js]: Routes after `handle: hit`, cannot set `dest` and must set `continue: true`

File: packages/now-routing-utils/test/index.spec.js [hunk 2]
```diff
@@ -54,6 +54,10 @@ describe('normalizeRoutes', () => {
...
+      { handle: 'miss' },
+      { src: '^/missed-me$', dest: '/api/missed-me' },
```
Comment (dav-is) [inline packages/now-routing-utils/test/index.spec.js]: Routes after `handle: miss` must set `check: true`, or if there's no `dest` `continue: true` instead

File: packages/now-routing-utils/test/index.spec.js (hunk 1 above)
Comment (styfle) [reply packages/now-routing-utils/test/index.spec.js]: Thanks, I updated the logic 👍

File: packages/now-routing-utils/test/index.spec.js (hunk 2 above)
Comment (styfle) [reply packages/now-routing-utils/test/index.spec.js]: Thanks, I updated the logic 👍

**Review by dav-is — APPROVED** [review]
:+1:

**Pull request merged**